frame/support/src/storage/child.rs

```diff
@@ -136,13 +136,66 @@ pub fn exists(child_info: &ChildInfo, key: &[u8]) -> bool {
 /// not make much sense because it is not cumulative when called inside the same block.
 /// Use this function to distribute the deletion of a single child trie across multiple
 /// blocks.
+#[deprecated = "Use `clear_storage` instead"]
 pub fn kill_storage(child_info: &ChildInfo, limit: Option<u32>) -> KillStorageResult {
 	match child_info.child_type() {
 		ChildType::ParentKeyId =>
 			sp_io::default_child_storage::storage_kill(child_info.storage_key(), limit),
 	}
 }
 
+/// Partially clear the child storage of each key-value pair.
+///
+/// # Limit
+///
+/// A *limit* should always be provided through `maybe_limit`. This is one fewer than the
```

Comment:
Are there any use cases where it makes sense to pass `None` to `maybe_limit`? If not, it sounds like we should just not make it an `Option`.

Reply:
Might be worth keeping the host functions general for now. Can't predict future uses quite yet.